src/object.c

```diff
@@ -398,16 +398,19 @@ void decrRefCount(robj *o) {
     }
 }
 
-/* See dismissObject() */
+/* See dismissObject(). sds is an exception. Instead of dismissing it
+ * with madvise(MADV_DONTNEED) we free it via the allocator. This has
+ * advantage that it allows the allocator to accumulate free buffers
+ * to free whole pages. While madvise is nop if the buffer is less
+ * than a page. The allocator overhead is minimal, because raw strings
+ * (sds) use the fast path. */
 void dismissSds(sds s) {
-    dismissMemory(sdsAllocPtr(s), sdsAllocSize(s));
+    sdsfree(s);
 }
 
 /* See dismissObject() */
 void dismissStringObject(robj *o) {
-    if (o->encoding == OBJ_ENCODING_RAW) {
-        dismissSds(o->ptr);
-    }
+    freeStringObject(o);
```

Comment:
BTW - why do we need to keep the freeStringObject here?

Reply:
```suggestion
    if (o->encoding == OBJ_ENCODING_RAW) {
        dismissSds(o->ptr);
    }
```

Reply:
We still use `freeStringObject` in `decrRefCount`. So this decision is extra call vs repeating code. I prefer the extra call as compiler can optimize that, while we avoid repeating the code.

Reply:
I am not sure what repeating code or extra call in this context. I just thought to keep this function as it was before since the change can only be done in dismissSds.
I do not see any issue with this change so I approved it. probably not worth the debate :)

Reply:
Apparently I misunderstood you. I took the change.